internal/server/data/access_key.go

```diff
@@ -94,18 +85,23 @@ func CreateAccessKey(db WriteTxn, accessKey *models.AccessKey) (body string, err
 		accessKey.ExpiresAt = time.Now().Add(time.Hour * 12).UTC()
 	}
 
-	if accessKey.Name == "" {
-		// set a default name for look-up and CLI usage
-		if accessKey.ID == 0 {
-			accessKey.ID = uid.New()
-		}
-
-		identityIssuedFor, err := GetIdentity(db, GetIdentityOptions{ByID: accessKey.IssuedFor})
+	if accessKey.IssuedForKind != models.AccessKeyForKindProvider {
+		identityIssuedFor, err := GetIdentity(db, GetIdentityOptions{ByID: accessKey.IssuedForID})
 		if err != nil {
 			return "", fmt.Errorf("key name from identity: %w", err)
 		}
 
-		accessKey.Name = fmt.Sprintf("%s-%s", identityIssuedFor.Name, accessKey.ID.String())
+		if identityIssuedFor.Name == models.InternalInfraConnectorIdentityName {
+			accessKey.IssuedForKind = models.AccessKeyForKindConnector
```

Comment:
Do we make use of this `KindConnector`, yet? Or could we introduce that if/when connector isn't an identity in the DB?

Reply:
Yeah it's not used at the moment. I see it as a way to get rid of the connector identity when we choose to do so, but it also felt incorrect to create the connector key as kind `user` in general. Im still open to doing this as a separate change though if you're worried about the lack of use.

Reply:
For me the question is more about if connector is the right `Kind` here. Maybe it should be `kind=organization` so that it's not specific to a connector. Any org admin would be able to manage `kind=organization`.

Reply:
Made this change to `kind=organization` for connector access keys.